ticket_migrate.py

```diff
@@ -0,0 +1,829 @@
+    """
+    if component in config.REPOSITORY_MAPPING:
+        return []
+
+    return [component]
+
+
+def labels_from_keywords(keywords: Union[str, None]):
+    """
+    Given the Trac `keywords` string, clean it up and parse it into a list.
+    """
+    if keywords is None:
+        return set()
+
+    keywords = keywords.replace(',', '')
+    keywords = keywords.split(' ')
+
+    allowed_keyword_labels = {
+        'design',
+        'easy',
+        'feature',
+        'happy-hacking',
+        'onboarding',
+        'password-management',
+        'perf-test',
+        'remote-management',
+        'salt',
+        'scp',
+        'security',
+        'tech-debt',
+        'twisted',
+        'ux',
+        }
+    typo_fixes = {'tech-dept': 'tech-debt', 'tech-deb': 'tech-debt'}
+
+    keywords = [typo_fixes.get(kw, kw) for kw in keywords if kw]
+    discarded = [kw for kw in keywords if kw not in allowed_keyword_labels]
+
+    if discarded:
+        print("Warning: discarded keywords:", discarded)
+
+    return {kw for kw in keywords if kw in allowed_keyword_labels}
+
+
+def labels_from_priority(priority):
+    """
+    Interpret None (missing) priority as Low.
+    """
+    if priority is None:
+        return 'priority-low'
+    return 'priority-{}'.format(priority.lower())
+
+
+def labels_from_status_and_resolution(status, resolution):
+    """
+    The resolution of a closed ticket is used, if there is one.
+    Otherwise, the status is used, if it is not "assigned", "new", or "closed".
+    """
+
+    if status == 'closed' and resolution:
+        return {resolution}
+
+    if status in ['in_work', 'needs_changes', 'needs_merge', 'needs_review']:
+        return {status}
+
+    return set()
+
+
+def parse_body(description):
+    """
+    Parses text with squiggly-bracketed or backtick-surrounded monospace.
+    Converts the squiggly brackets to backtick brackets.
+    """
+    if not description:
+        return ''
+
+    def found(index):
+        """Return true if an index represents a found position in a string."""
+        return index != -1
+
+    def is_first_index(a, b):
+        """
+        Returns true if index a occurs before b, or if b does not exist.
+        """
+        return found(a) and (not found(b) or b > a)
+
+    min_backtick = description.find('`')
+    min_squiggly = description.find('{{{')
+
+    if is_first_index(min_squiggly, min_backtick):
+        return (
+            convert_issue_content(description[:min_squiggly]) +
+            parse_squiggly(description[min_squiggly:])
+            )
+
+    if is_first_index(min_backtick, min_squiggly):
+        return (
+            convert_issue_content(description[:min_backtick]) +
+            parse_backtick(description[min_backtick:])
+            )
+
+    return convert_issue_content(description)
+
+
+def convert_issue_content(text):
+    """
+    Convert TracWiki text to GitHub Markdown.
+    Ignore included images.
+    """
+    return convert(text, '')
```

Comment:
I am not sure why we have empty string here.

When possible, is best to call the methods with named arguments

```suggestion
    return convert(text, multilines=False)
```

Reply:
> best to call the methods with named arguments

This is `base_path`; point taken. I use an empty string because we don't plan to support image and wiki migration in this PR, yet something has to be provided as an argument.